api/src/main/java/io/opentelemetry/common/AttributeConsumer.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+/** Convenience interface for consuming {@link ReadableAttributes}. */
```

Comment:
I think for these sort of interfaces, at least until we're on Java 8 someday, we can add a comment like. `This is a FunctionalInterface and will not have methods added to it - it is safe to use as a lambda without chance of API breakage. The reason we don't add the annotation is we target Java 7.`

Reply:
A good point. But, if I'm going ahead with changing the API for #1580 , this one won't be so nice and simple any more, but have lots of methods on it. The LabelConsumer is a different story, though, as I think it will stay simple.